Add VAL_TRUE and VAL_FALSE macros for use in doxygen comments

nyalldawson opened the pull request:
For the c++ api dox these expand to "\c true" and "\c false" (the \c directive indicates a code literal value), and for sipify/Python they expand to ``True`` and ``False`` (`` is sphinx annotation for literal values)

Makes for nicer dox for both c++ and Python!

nyalldawson committed:
Add TRUE and FALSE macros for use in doxygen comments

For the c++ api dox these expand to "\c true" and "\c false" (the
\c directive indicates a code literal value), and for sipify/Python
they expand to ``True`` and ``False`` (`` is sphinx annotation
for literal values)

Makes for nicer dox for both c++ and Python!

m-kuhn commented on scripts/sipify.pl:
```diff
@@ -456,6 +457,13 @@ sub fix_constants {
 
+sub replace_macros {
+    my $line = $_[0];
+    $line =~ s/\bVAL_TRUE\b/``True``/g;
```
What about the following, it would read a bit easier in the original header files.

```suggestion
    $line =~ s/\b\\c true\b/``True``/g;
```

nyalldawson replied:
Do you mean instead of the macros?

m-kuhn replied:
Yes, exactly. The proposal is to document C++ and derive the Python part (instead of documenting in a meta-language)

nyalldawson replied:
I did think about this overnight, but I don't really like it. For a start it makes reviewing PRs and identifying missed raw true/false strings a bit harder, but more importantly, it would make it very difficult to find places missing the \c prefix (you'd need to use a negative lookbehind regex, which aren't widely supported). I'd also like to expand on this and add others like NULLPTR. I'm also slightly worried that \c isn't the correct annotation to use for doxygen, so i'd like it to be easy to revise this in future.

What about just an all caps TRUE/FALSE string instead?

m-kuhn replied:
Good points.
`TRUE`, `FALSE` and `NULLPTR` sound readable.

nyalldawson replied:
Sweet! Got any other suggestions?

I'm wondering if numeric constants should also be treated specially, e.g. "returns 0 when ..."

m-kuhn replied:
"Returns ``0`` when" should work fine for C++ and Python. And it's probably worth considering upgrading return values to an enum in such cases.

Anyway, true, false and nullptr are probably the most important ones. Other things to look into would be more cross-linking inside the docs and adding code samples, so nothing that can be done as easy as the ones here.

m-kuhn commented on src/core/qgsvectorlayer.h:
```diff
@@ -406,12 +406,12 @@ class CORE_EXPORT QgsVectorLayer : public QgsMapLayer, public QgsExpressionConte
       {}
 
-      //! Sets to true if the default layer style should be loaded
+      //! Sets to VAL_TRUE if the default layer style should be loaded
```
Unrelated suggestion

```suggestion
      //! Set to VAL_TRUE if the default layer style should be loaded
```

nyalldawson committed:
Nullptr consistency in dox

nyalldawson committed:
Bulk doxygen TRUE/FALSE literal conversion

nyalldawson committed:
Sipify

nyalldawson committed:
astyle

nyalldawson committed:
Retval -> returns

nyalldawson committed:
Typo, remove unused .sip files

nyalldawson committed:
Add NULLPTR macro for use in doxygen comments

For the c++ api dox this expands to "\c nullptr" (the
\c directive indicates a code literal value), and for sipify/Python
it expands to ``None`` (`` is sphinx annotation for literal values)

Makes for nicer dox for both c++ and Python!

nyalldawson committed:
Fix doxygen warning

nyalldawson committed:
Update sipify expected results

nyalldawson committed:
Fix doxygen input filter
